From b0dbeb18ee36a2d17f0f039a739e2e022f305f62 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Tue, 1 Dec 2015 16:27:34 +0100
Subject: [PATCH 3/3] Minimize device tree for S Core eval board

The device tree for the S Core evaluation board is minimized. There are
only the main components in which contains uart1, sd3 and gpio led on
the S Core. Although the usage of the score include device tree is used.

Ticket #805
---
 arch/arm/boot/dts/imx6sx-score.dts | 382 ++++---------------------------------
 1 file changed, 32 insertions(+), 350 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-score.dts b/arch/arm/boot/dts/imx6sx-score.dts
index 576e164..051851e 100644
--- a/arch/arm/boot/dts/imx6sx-score.dts
+++ b/arch/arm/boot/dts/imx6sx-score.dts
@@ -9,7 +9,8 @@
 /dts-v1/;
 
 #include <dt-bindings/input/input.h>
-#include "imx6sx.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include "imx6sx-score.dtsi"
 
 / {
 	model = "sitec systems S Core Eval-Board";
@@ -20,26 +21,22 @@
 	};
 
 	memory {
-		reg = <0x80000000 0x40000000>;
+		reg = <0x80000000 0x20000000>;
 	};
 
+    leds {
+        compatible = "gpio-leds";
+        alive {
+            label = "Alive";
+            gpios = <&gpio3 19 GPIO_ACTIVE_HIGH>;
+            linux,default-trigger = "heartbeat";
+        };
+    };
 
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <0>;
-
-		reg_5p0: regulator@4 {
-			compatible = "regulator-fixed";
-			reg = <4>;
-			pinctrl-names = "default";
-			pinctrl-0 = <&pinctrl_5pe>;
-			regulator-name = "5p0";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			gpio = <&gpio3 18 GPIO_ACTIVE_HIGH>;
-			regulator-boot-on;
-		};
 	};
 };
 
@@ -58,287 +55,48 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
-	fsl,arm-soc-shared = <1>;
+    arm-supply = <&reg_arm>;
+    soc-supply = <&reg_soc>;
 };
 
 &gpc {
 	/* use ldo-bypass, u-boot will check it and configure */
-	fsl,ldo-bypass = <1>;
-};
-
-&fec1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_enet1>;
-	phy-mode = "rgmii";
-	status = "okay";
-};
-
-&fec2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_enet2>;
-	phy-mode = "rmii";
-	status = "okay";
-};
-
-&flexcan2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan2>;
-	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_LOW>;
-	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
-	status = "okay";
+	fsl,ldo-bypass = <0>;
 };
 
 &uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1>;
-	status = "okay";
-};
-
-&uart2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart2>;
-	status = "okay";
-};
-
-&uart6 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart6>;
-	fsl,uart-has-rtscts;
-	status = "okay";
-};
-
-&usbotg1 {
-	dr_mode = "host";
-	status = "okay";
-};
-
-&usdhc2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc2>;
-	bus-width = <4>;
-	non-removable;
-	no-1-8-v;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_uart1>;
+    status = "okay";
 };
 
 &usdhc3 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc3>;
-	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
 	bus-width = <4>;
 	cd-gpios = <&gpio7 6 GPIO_ACTIVE_HIGH>;
 	wp-gpios = <&gpio7 7 GPIO_ACTIVE_HIGH>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	status = "okay";
-};
-
-&i2c1 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c1>;
-	status = "okay";
-};
-
-&i2c4 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c4>;
 	status = "okay";
-
-	pmic: pfuze200@08 {
-		compatible = "fsl,pfuze200";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1ab {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3a {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3b_reg: sw3b {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vgen1 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen2_reg: vgen2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen3_reg: vgen3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-			};
-
-			vgen4_reg: vgen4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vgen5 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vgen6 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
 };
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_can_gpios &pinctrl_wlan_ctrl &pinctrl_pctrl>;
+    pinctrl-0 = <&pinctrl_gpio>;
 
 	imx6x-score {
-		pinctrl_pctrl: power-ctrl {
-			fsl,pins = <
-				/* 3P3_1_EN */
-				MX6SX_PAD_LCD1_DATA22__GPIO3_IO_23	0x4018
-				/* 3P3_2_DIS */
-				MX6SX_PAD_LCD1_DATA21__GPIO3_IO_22	0x4018
-				/* 1P8_EN */
-				MX6SX_PAD_LCD1_VSYNC__GPIO3_IO_28	0x4018
-				/* 4P0_DIS */
-				MX6SX_PAD_LCD1_DATA16__GPIO3_IO_17	0x4018
-				/* PWRGD_4P */
-				MX6SX_PAD_LCD1_ENABLE__GPIO3_IO_25	0x4018
-			>;
-		};
-
-		pinctrl_5pe: power_5P0_enable {
-			fsl,pins = <
-				/* 5P_EN */
-				MX6SX_PAD_LCD1_DATA17__GPIO3_IO_18	0x4018
-			>;
-		};
-
-		pinctrl_enet1: enet1grp {
-			fsl,pins = <
-				MX6SX_PAD_RGMII1_TD0__ENET1_TX_DATA_0	0xa0b1
-				MX6SX_PAD_RGMII1_TD1__ENET1_TX_DATA_1	0xa0b1
-				MX6SX_PAD_RGMII1_TD2__ENET1_TX_DATA_2	0xa0b1
-				MX6SX_PAD_RGMII1_TD3__ENET1_TX_DATA_3	0xa0b1
-				MX6SX_PAD_RGMII1_TX_CTL__ENET1_TX_EN	0xa0b1
-				MX6SX_PAD_RGMII1_TXC__ENET1_TX_ER		0xa0b1
-				MX6SX_PAD_RGMII1_RD0__ENET1_RX_DATA_0	0x3081
-				MX6SX_PAD_RGMII1_RD1__ENET1_RX_DATA_1	0x3081
-				MX6SX_PAD_RGMII1_RD2__ENET1_RX_DATA_2	0x3081
-				MX6SX_PAD_RGMII1_RD3__ENET1_RX_DATA_3	0x3081
-				MX6SX_PAD_RGMII1_RX_CTL__ENET1_RX_EN	0x3081
-				MX6SX_PAD_RGMII1_RXC__ENET1_RX_ER		0x3081
-				MX6SX_PAD_ENET1_RX_CLK__ENET1_RX_CLK	0x91
-				MX6SX_PAD_ENET1_TX_CLK__ENET1_TX_CLK	0x91
-				MX6SX_PAD_ENET1_MDIO__ENET1_MDIO		0xa0b1
-				MX6SX_PAD_ENET1_MDC__ENET1_MDC			0xa0b1
-			>;
-		};
-
-		pinctrl_enet2: enet2grp {
-			fsl,pins = <
-				MX6SX_PAD_RGMII2_RX_CTL__ENET2_RX_EN	0x3081
-				MX6SX_PAD_RGMII2_RD0__ENET2_RX_DATA_0	0x3081
-				MX6SX_PAD_RGMII2_RD1__ENET2_RX_DATA_1	0x3081
-				MX6SX_PAD_RGMII2_TX_CTL__ENET2_TX_EN	0xa0b1
-				MX6SX_PAD_RGMII2_TD0__ENET2_TX_DATA_0	0xa0b1
-				MX6SX_PAD_RGMII2_TD1__ENET2_TX_DATA_1	0xa0b1
-				MX6SX_PAD_ENET2_CRS__ENET2_CRS			0x80000000
-				MX6SX_PAD_ENET2_TX_CLK__ENET2_REF_CLK2  0x91
-			>;
-		};
-
-		pinctrl_flexcan1: flexcan1grp {
-			fsl,pins = <
-				MX6SX_PAD_KEY_COL2__CAN1_TX			0x1b020
-				MX6SX_PAD_KEY_ROW2__CAN1_RX			0x1b020
-			>;
-		};
-
-		pinctrl_flexcan2: flexcan2grp {
-			fsl,pins = <
-				MX6SX_PAD_KEY_COL3__CAN2_TX			0x1b020
-				MX6SX_PAD_KEY_ROW3__CAN2_RX			0x1b020
-			>;
-		};
-
-		pinctrl_can_gpios: can-gpios {
-			fsl,pins = <
-				/* CAN1_EN */
-				MX6SX_PAD_KEY_ROW4__GPIO2_IO_19		0x17059
-				/* CAN1_TERM */
-				MX6SX_PAD_LCD1_DATA21__GPIO3_IO_22	0x17059
-				/* CAN2_EN */
-				MX6SX_PAD_KEY_COL4__GPIO2_IO_14		0x17059
-				/* CAN2_TERM */
-				MX6SX_PAD_GPIO1_IO08__GPIO1_IO_8	0x17059
-			>;
-		};
-
-		pinctrl_i2c1: i2c1grp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
-				MX6SX_PAD_GPIO1_IO00__I2C1_SCL 0x4001b8b1
-			>;
-		};
-
-		pinctrl_i2c4: i2c4grp {
-			fsl,pins = <
-				MX6SX_PAD_CSI_DATA07__I2C4_SDA 0x4001b8b1
-				MX6SX_PAD_CSI_DATA06__I2C4_SCL 0x4001b8b1
-			>;
-		};
+        pinctrl_gpio: gpiogrp {
+            fsl,pins = <
+                MX6SX_PAD_LCD1_DATA18__GPIO3_IO_19  0x17059
+                MX6SX_PAD_LCD1_DATA19__GPIO3_IO_20  0x17059
+            >;
+        };
+
+        pinctrl_i2c1: i2c1grp {
+            fsl,pins = <
+                MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
+                MX6SX_PAD_GPIO1_IO00__I2C1_SCL 0x4001b8b1
+            >;
+        };
 
 		pinctrl_uart1: uart1grp {
 			fsl,pins = <
@@ -347,60 +105,6 @@
 			>;
 		};
 
-		pinctrl_uart2: uart2grp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO06__UART2_TX		0x1b0b1
-				MX6SX_PAD_GPIO1_IO07__UART2_RX		0x1b0b1
-			>;
-		};
-
-		pinctrl_uart6: uart6grp {
-			fsl,pins = <
-				MX6SX_PAD_KEY_ROW0__UART6_CTS_B		0x1b0b1
-				MX6SX_PAD_KEY_COL0__UART6_RTS_B		0x1b0b1
-				MX6SX_PAD_KEY_COL1__UART6_TX		0x1b0b1
-				MX6SX_PAD_KEY_ROW1__UART6_RX		0x1b0b1
-			>;
-		};
-
-		pinctrl_usb_otg1: usbotg1grp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO09__GPIO1_IO_9	0x10b0
-			>;
-		};
-
-		pinctrl_usb_otg1_id: usbotg1idgrp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO10__ANATOP_OTG1_ID	0x17059
-			>;
-		};
-
-		pinctrl_usb_otg2: usbot2ggrp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO12__GPIO1_IO_12	0x10b0
-			>;
-		};
-
-		pinctrl_wlan_ctrl: wlan_ctrl {
-			fsl,pins = <
-				/* WIFI EN */
-				MX6SX_PAD_GPIO1_IO10__GPIO1_IO_10	0x4018
-				/* WIFI Res */
-				MX6SX_PAD_GPIO1_IO11__GPIO1_IO_11	0x4018
-			>;
-		};
-
-		pinctrl_usdhc2: usdhc2grp {
-			fsl,pins = <
-				MX6SX_PAD_SD2_CMD__USDHC2_CMD		0x17059
-				MX6SX_PAD_SD2_CLK__USDHC2_CLK		0x10059
-				MX6SX_PAD_SD2_DATA0__USDHC2_DATA0	0x17059
-				MX6SX_PAD_SD2_DATA1__USDHC2_DATA1	0x17059
-				MX6SX_PAD_SD2_DATA2__USDHC2_DATA2	0x17059
-				MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17059
-			>;
-		};
-
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
@@ -413,27 +117,5 @@
 				MX6SX_PAD_SD3_DATA5__GPIO7_IO_7		0x17059 /* WP */
 			>;
 		};
-
-		pinctrl_usdhc3_100mhz: usdhc3grp-100mhz {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170b9
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100b9
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170b9
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170b9
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170b9
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170b9
-			>;
-		};
-
-		pinctrl_usdhc3_200mhz: usdhc3grp-200mhz {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170f9
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100f9
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170f9
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170f9
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170f9
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170f9
-			>;
-		};
 	};
 };
-- 
1.9.1

